packages/ui/src/primitives/connection.tsx

```diff
@@ -3,20 +3,36 @@ import * as React from 'react';
 
 import { Spinner } from './spinner';
 
-export const Root = React.forwardRef<HTMLDivElement, React.HTMLAttributes<HTMLDivElement>>(function Root(
-  { children, className, ...props },
-  forwardedRef,
-) {
+function getColumnCount<T>(items: T[], maxCols: number = 6): number {
+  const numRows = Math.ceil(items.length / maxCols);
+  const itemsPerRow = Math.ceil(items.length / numRows);
+  const rows: T[][] = Array.from({ length: numRows }, () => []);
+  let currentArrayIndex = 0;
+  for (const item of items) {
+    rows[currentArrayIndex].push(item);
+    if (rows[currentArrayIndex].length === itemsPerRow) {
+      currentArrayIndex++;
+    }
+  }
+  return Math.max(...rows.map(row => row.length));
+}
+
+export const Root = React.forwardRef<
+  HTMLDivElement,
+  React.HTMLAttributes<HTMLDivElement> & {
+    columns?: number;
+  }
+>(function Root({ children, className, ...props }, forwardedRef) {
+  const cols = getColumnCount(React.Children.toArray(children));
   return (
     <div
       ref={forwardedRef}
       {...props}
       className={cn(
-        '[--cl-connection-gap:theme(spacing.2)]',
-        '[--cl-connection-cols:2]',
-        'flex gap-[--cl-connection-gap] flex-wrap items-center justify-center',
+        'flex items-center justify-center flex-wrap [--cl-connection-gap:theme(spacing.2)] gap-y-[--cl-connection-gap] -mx-[calc(var(--cl-connection-gap)/2)]',
         className,
       )}
+      style={{ '--cl-connection-cols': cols } as React.CSSProperties}
```

Comment:
Transformer currently doesn't account for template literals. Will follow up a fix.

Reply:
love that we're collecting different uses and making the transformer better each time 📈